src/constraints/field.rs

```diff
@@ -220,6 +220,61 @@ fn equal_cells<B: Backend>(
     }
 }
 
+/// Returns 1 if lhs != rhs, 0 otherwise
+pub fn not_equal<B: Backend>(
+    compiler: &mut CircuitWriter<B>,
+    lhs: &Var<B::Field, B::Var>,
+    rhs: &Var<B::Field, B::Var>,
+    span: Span,
+) -> Var<B::Field, B::Var> {
+    assert_eq!(lhs.len(), rhs.len());
+
+    let one = B::Field::one();
+
+    if lhs.len() == 1 {
+        let diff = sub(compiler, &lhs[0], &rhs[0], span);
+        let is_zero = is_zero_cell(compiler, &diff[0], span);
+        return boolean::not(compiler, &is_zero[0], span);
+    }
+
+    let acc = compiler.backend.add_constant(
+        Some("start accumulator at 1 for the inequality check"),
+        one,
+        span,
+    );
+    let mut acc = Var::new_var(acc, span);
+
+    for (l, r) in lhs.cvars.iter().zip(&rhs.cvars) {
+        let diff = sub(compiler, l, r, span);
+        let res = is_zero_cell(compiler, &diff[0], span);
+        let not_res = boolean::not(compiler, &res[0], span);
+        acc = boolean::and(compiler, &not_res[0], &acc[0], span);
+    }
```

Comment:
btw there's probably a faster way to implement that, just sum the results of `is_zero_cell`, and then do a `not(is_zero(acc))` at the end